src/ansys/simai/core/data/projects.py

```diff
@@ -77,6 +106,19 @@ def delete(self) -> None:
         """Delete the project."""
         self._client._api.delete_project(self.id)
 
+    def is_trainable(self) -> bool:
+        """Check if the project meets the prerequisites to be trained."""
+        tt = self._client._api.is_project_trainable(self.id)
+        return IsTrainableInfo(**tt)
+
+    def get_variables(self) -> dict:
+        """Get the available variables for the model's input/output."""
+        variables = self.sample.fields.get("extracted_metadata")
+        data = {}
+        for key in variables:
+            data |= {key: [var.get("name") for var in variables.get(key).get("fields")]}
+        return data
```

Comment:
1. Seems this will crash if the project has no sample, maybe we should instead return `None`
2. Let's make the code easier to read:
```suggestion
    def get_variables(self) -> dict:
        """Get the available variables for the model's input/output."""
        sample_metadata = self.sample.fields.get("extracted_metadata")
        vars = {}
        for location, located_data in sample_metadata.items():
            local_fields = located_data.get("fields", [])
            vars[location] = [field.get("name") for field in local_fields]
```
3. IDK what your spec looks like on this one but IMO `variable` isn't a great name:
    * In the SDK everything is a variable, IMO it's confusing
    * Internally it's called a field (following VTK's naming on this one)
    * We have features planned which use `field` in the name

Reply:
1. good catch! I added a check.
2. you didn't like the merge operator? 😢 I updated the code.
3. I try to follow the same naming as in the FE so it can be easier for the user to digest. The name there is `variables`
![image](https://github.com/ansys/pysimai/assets/10939450/a0919c47-d635-4d64-9eda-e40e96d75197)
If you have another idea, let me know.

Reply:
good point, I removed it.